### ISSUE TITLE
IIIF Server Base URL Settings field fails validity check within Settings Form

### ISSUE THREAD

**Issue by tomadams** [esmero/archipelago-deployment#61]
Following the installation directions for Installing Archipelago on Ubuntu (actually on a RHEL 7.6) for Beta3, seemed to have encountered the same issue as with Beta2. The difficulty appears when setting up Archipelago on a remote server (i.e. you are not using "http://localhost:8183/iiif/2/" as the IIIF Server base URL, but instead need to specify an actual base name.)

One should be able to specify the "Base URL of your IIIF Media Server public accesible [accessible] from the Outside World." from the IIIF Server Settings Form. This is not the case, since the form validation does not accept a valid URL of the IIIF server. A guess, is that the form validation for that field, might be querying for a IIIF media server at that URL, but it is querying on an INTERNAL docker network and not on the outward looking (non-virtual) network. The example given below, the URL provided does resolve in the web browser, but is not seen as valid in the form.

![image](https://user-images.githubusercontent.com/1951133/89186948-8b631e80-d56a-11ea-82d8-2574d0ba9a1d.png)

My current work around, is to modify the configuration files by following the steps located at: https://groups.google.com/d/msg/archipelago-commons/N5D0M6h7DUI/6xLYWfKUAwAJ
But that might be asking to much, for a first time Archipelago user.

**Comment by DiegoPino** [comment]
@tomadams thanks for reporting this. 

We have a few public facing AWS/Digital Ocean Servers with publicly exposed IIIF servers that were setup using the form without issues, so i wonder if the problem is other or i missed something in the code.

You running Beta3?

The validation that happens in code is actually trying to parse the response (of cantaloupe)
See https://github.com/esmero/format_strawberryfield/blob/8.x-1.0-beta3/src/Tools/IiifUrlValidator.php#L97-L120

So, any chance that the URL can not be seen as "publicly" accessible from the web server at the moment of the setup? Or maybe your IIIF server is blocked by firewall/rule?

Could you check your Drupal  logs? if validation is not possible, Archipelago will log some more finer descriptive error there.

Would love to be able to test with your real domain name (any chance you can share it privately to see the response?)

Let me know. If nothing helps i can iterate over the code again and see where the error is and apply a quick patch, but i feel the problem could be somewhere else so better to do a double check.

**Comment by giancarlobi** [comment]
Hi Tom,

I'm on Archipelago beta 3 and an installation without docker. I'm using 
external Cantaloupe on another server and settings works well for me.

My settings:

![image](https://user-images.githubusercontent.com/4232178/89193419-3e9d3900-d5a6-11ea-9b7f-134913dbaf80.png)


IMHO I think you need reverse proxy on Archipelago to proxy external 
Cantaloupe server to avoid inter-domains browser error.

On my archipelago I'm using something like this into apache conf:

         AllowEncodedSlashes On
         ProxyRequests Off
         ProxyPreserveHost On
         <Proxy *>
                 Order deny,allow
                 Allow from all
         </Proxy>


         RequestHeader set X-Forwarded-Path /iiif-server
         ProxyPass /iiif-server http://[CantaloupeIP]:8182 nocanon
         ProxyPassReverse /iiif-server http://[CantaloupeIP]:8182

Happy if you need more help

Giancarlo

**Comment by tomadams** [comment]
I am running beta3.

Looking in the logs, there is the following error message:
We could not contact the IIIF Server at http://zzzzzz.edu:8183/iiif/2 with error: cURL error 6: Could not resolve host: zzzzzz.edu (see https://curl.haxx.se/libcurl/c/libcurl-errors.html)

Looks like my server has a DNS issue. Ran dig and nslookup on the server, and that machine was not able to resolve it's own name.  My local PC, however, is able to resolve that name - it uses a separate DNS server.

Marking this issue as closed. It was a local machine configuration problem.

Diego - Thanks for indicating there might be a more detailed error message in the logs, which was the case.

Giancarlo - Thanks for your reply. I believe your information is going to be useful shortly.

-Tom

**Comment by DiegoPino** [comment]
Thanks @tomadams! Please don't hesitate to reach out anytime. Will also share soon the SSL config for NGINX here too (another issue, i recommend you subscribe/watch this repo if you are not right now), in case you are planning on going that route.

best!
